astropy/units/tests/test_format.py

```diff
@@ -348,17 +346,12 @@ def test_roundtrip(self, unit):
 
 
 class TestRoundtripVOUnit(RoundtripBase):
-    format_ = "vounit"
-    deprecated_units = u_format.VOUnit._deprecated_units
+    format_ = u_format.VOUnit
 
     @pytest.mark.parametrize(
         "unit",
-        [
-            unit
-            for unit in u_format.VOUnit._units.values()
-            if (isinstance(unit, UnitBase) and not isinstance(unit, PrefixUnit))
-        ],
-        ids=lambda x: str(x),
+        [u for u in u_format.VOUnit._units.values() if not isinstance(u, PrefixUnit)],
```

Comment:
I had to do a double take, worried we're overriding the module. I'd prefer `v` (for `value`) or just a generic `_`, but also fine to leave as is, since clearly it stays in scope.

Reply:
Aside: I wonder if the parametrized test is still the most logical: if one were to write this as a loop inside `test_roundtrip`, one could access `cls.format_` - i.e., the test would be the same for all formats.

Reply:
1. If a test contains asserts in a loop and one of those asserts fails then that stops the loop and prevents the following asserts from being evaluated. But if one assert in a parametrized test fails then the other parameters will still be evaluated (unless the user has _chosen_ to call `pytest` with the [`--exitfirst`](https://docs.pytest.org/en/stable/reference/reference.html#command-line-flags) option or something like that). 
2. Parametrized tests can be more informative if they fail or if the user runs [`pytest` with increased verbosity](https://docs.pytest.org/en/stable/how-to/output.html#verbosity) thanks to [customizable ids](https://docs.pytest.org/en/stable/example/parametrize.html#different-options-for-test-ids).
3. The parametrizations of a parametrized test can be selected individually. For example, see what happens with `pytest --verbose -k 'TestRoundtrip and kg' -- astropy/units/tests/test_format.py`.

Reply:
Good point about users being able to pick a specific test. Sadly, I don't think one can parametrize on a class variable... Anyway, let's keep it as you have it.

Reply:
When I ran `pytest --verbose -k 'TestRoundtrip and kg' -- astropy/units/tests/test_format.py` I noticed that `TestRoundtripGeneric.test_roundtrip()` seems to be called twice with `kg`. The reason is that there is `u.kg` and `u.kilogram` and `TestRoundtripGeneric` tests both. At the very least the test ids should be updated to make the difference clear.